### ISSUE TITLE
"Error: destroy namespace: namespace1.0 failed to enable for zeroing, continuing" observed with destroy-namespace operation

### ISSUE THREAD

**Issue by yizhanglinux** [pmem/ndctl#164]
Hi
I found this Error log when I do destroy-namespace operation, and it was introduced with commit [2], pls help check it.

[1]
```
# ndctl list -i
[
  {
    "dev":"namespace1.0",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":4096,
    "state":"disabled"
  },
  {
    "dev":"namespace1.1",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":512,
    "state":"disabled"
  },
  {
    "dev":"namespace3.0",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":512,
    "state":"disabled"
  },
  {
    "dev":"namespace0.1",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":512,
    "state":"disabled"
  },
  {
    "dev":"namespace0.0",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":4096,
    "state":"disabled"
  },
  {
    "dev":"namespace2.0",
    "mode":"raw",
    "size":0,
    "uuid":"00000000-0000-0000-0000-000000000000",
    "sector_size":512,
    "state":"disabled"
  }
]
# ./ndctl/ndctl destroy-namespace -f all
  Error: destroy namespace: namespace1.0 failed to enable for zeroing, continuing

  Error: destroy namespace: namespace1.1 failed to enable for zeroing, continuing

  Error: destroy namespace: namespace3.0 failed to enable for zeroing, continuing

  Error: destroy namespace: namespace0.1 failed to enable for zeroing, continuing

  Error: destroy namespace: namespace0.0 failed to enable for zeroing, continuing

  Error: destroy namespace: namespace2.0 failed to enable for zeroing, continuing

destroyed 0 namespaces

```
[2]
```
commit 46654c2d60b70fdc326fd4c1e83da7667121b7bf (HEAD)
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Sat Feb 29 12:21:19 2020 -0800

    ndctl/namespace: Always zero info-blocks
```

**Comment by JeffMoyer** [comment]
It should be sufficient to skip zero-sized namespaces in zero_info_block.

**Comment by djbw** [comment]
Agree, it really did fail to zero info blocks there because there was nothing to zero, but probably better to just be quiet.

**Comment by yizhanglinux** [comment]
Yes, fixed by below commit, close this issue now.

```
commit fb13dfb8d84c4f0a749665c8f07179450b199f3e
Author: Jeff Moyer <jmoyer@redhat.com>
Date:   Tue Feb 9 16:51:53 2021 -0500

    zero_info_block: skip seed devices
    
    Currently, ndctl destroy-namespace -f all will output errors of the
    form:
    
      Error: destroy namespace: namespace0.0 failed to enable for zeroing, continuing
    
    for any zero-sized namespace.  That particular namespace looks like this:
    
      {
        "dev":"namespace0.0",
        "mode":"raw",
        "size":0,
        "uuid":"00000000-0000-0000-0000-000000000000",
        "sector_size":512,
        "state":"disabled"
      }
    
    This patch skips over namespaces with size=0 when zeroing out info
    blocks.
    
    Fixes: 46654c2d60b70 ("ndctl/namespace: Always zero info-blocks")
    Reported-by: Zhang Yi <yizhan@redhat.com>
    Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
    Signed-off-by: Vishal Verma <vishal.l.verma@intel.com>
    Link: https://lore.kernel.org/r/x49r1lohpty.fsf@segfault.boston.devel.redhat.com
```
